AppRentCar/templates/available_cars.html

```diff
@@ -1,24 +1,86 @@
         <div class="row">
             {% for car in object_list %}
-            <div class="col-md-4 mb-4">
-                <div class="card">
-                    <img src="{% url 'images' car.avatar %}" height="150" width="250">
-                    <div class="card-body">
-                        <a href="{% url 'car_detail' car.id %}">{{ car.brand }} {{ car.model }} <br>{{ car.year }}</br>
-                        </a>
+                <div class="col-md-4 mb-4">
+                    <div class="card">
+                        <img src="{% url 'images' car.avatar %}" height="150" width="250">
+                        <div class="card-body">
+                            <a href="{% url 'car_detail' car.id %}">{{ car.brand }} {{ car.model }} <br>{{ car.year }}</br>
+                            </a>
+                        </div>
                     </div>
                 </div>
-            </div>
-        {% endfor %}
-
+            {% endfor %}
+        </div>
+    </div>
 {% else %}
     <p>Brak dostępnych samochodów w podanym terminie.</p>
 {% endif %}
+<script>
+$(function() {
+    $("#start-date").datepicker();
+    $("#end-date").datepicker();
+
+    $("#car-filter-form").submit(function(event) {
+        event.preventDefault();
+
+        var startDate = $("#start-date").datepicker("getDate");
+        var endDate = $("#end-date").datepicker("getDate");
+
+        var engineType = $("#id_engine").val();
+        var driveType = $("#id_drive").val();
+        var transmission = $("#id_transmission").val();
+        var carsType = $("#id_cars_type").val();
+
+        if (startDate && endDate) {
+            var formattedStartDate = $.datepicker.formatDate("yy-mm-dd", startDate);
+            var formattedEndDate = $.datepicker.formatDate("yy-mm-dd", endDate);
+
+            var filterURL = "/available-cars/?start_date=" + formattedStartDate + "&end_date=" + formattedEndDate;
+            if (engineType) {
+                filterURL += "&engine=" + encodeURIComponent(engineType);
+            }
+            if (driveType) {
+                filterURL += "&drive=" + encodeURIComponent(driveType);
+            }
+            if (transmission) {
+                filterURL += "&transmission=" + encodeURIComponent(transmission);
+            }
+            if (carsType) {
+                filterURL += "&cars_type=" + encodeURIComponent(carsType);
+            }
+
+            window.location.href = filterURL;
+        }
+    });
+
+    var urlParams = new URLSearchParams(window.location.search);
+    var startParam = urlParams.get('start_date');
+    var endParam = urlParams.get('end_date');
 
+    if (!$("#start-date").val() && startParam) {
+        var startDate = new Date(startParam);
+        $("#start-date").datepicker("setDate", startDate);
+    }
+    if (!$("#end-date").val() && endParam) {
+        var endDate = new Date(endParam);
+        $("#end-date").datepicker("setDate", endDate);
+    }
+});
+</script>
```

Comment:
możliwe, że to mógłbyś zastąpić jakimś formularzem GET który będzie dodawał do query parameter, warto spojrzeć na bibliotekę django-filter